iceoryx_hoofs/include/iceoryx_hoofs/cxx/optional.hpp

```diff
@@ -224,6 +230,8 @@ class optional : public FunctionalInterface<optional<T>, T, void>
     const T&& value() const&& noexcept;
 
   private:
+    // safe access is guaranteed since the array is wrapped inside the optional
```

Comment:
Maybe also add a todo #1196 here and state that this will be replaced with an uninitialized array